pgjdbc/src/main/java/org/postgresql/util/ReaderInputStream.java

```diff
@@ -0,0 +1,187 @@
+package org.postgresql.util;
+
+//#if mvn.project.property.postgresql.jdbc.spec >= "JDBC4.1"
+import static java.nio.charset.StandardCharsets.UTF_8;
+//#endif
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.CharsetEncoder;
+import java.nio.charset.CoderResult;
+import java.nio.charset.MalformedInputException;
+
+public class ReaderInputStream extends InputStream {
```

Comment:
java.io.InputStreamReader takes a byte stream and a Charset as input and outputs a UTF-16 char stream. ReaderInputStream takes a UTF-16 char stream as input and outputs a UTF-8 byte stream. I hardcoded UTF-8 as the target charset because pgjdbc always uses UTF-8 for front end<->back end communication.